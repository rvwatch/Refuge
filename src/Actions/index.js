export const addUser = user => ({
  type: 'ADD_USER',
  user
});
