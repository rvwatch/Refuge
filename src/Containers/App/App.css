.App-header {
  overflow: hidden;
  height: 40px;
  padding: 20px;
  background: rgb(68,161,167);
  background: -moz-linear-gradient(45deg, rgba(68,161,167,1) 0%, rgb(114, 255, 203) 100%);
  background: -webkit-linear-gradient(45deg, rgba(68,161,167,1) 0%,rgba(114,255,203,1) 100%);
  background: linear-gradient(45deg, rgba(68,161,167,1) 0%,rgba(114,255,203,1) 100%);
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  position: relative;
}

.logo-wrap {
  display: block;
  border-radius: 100%;
  border: 6px solid #83B8F4;
  background-color: #fff;
  width: 200px;
  height: 200px;
  left: -60px;
  top: -120px;
  position: relative;
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.75);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.75);
}

.logo {
  width: 80px;
  position: absolute;
  right: 45px;
  top: 108px;
  transition: all .3s;
}

.logo-wrap:hover .logo {
  width: 90px;
  right: 40px;
  top: 103px;
}

.account {
  position: absolute;
  top: 25px; 
  right: 30px;
  color: #4e79a9
}

.account .logout-btn {
  text-decoration: underline;
  margin-right: 20px;
  color: #4e79a9;
  cursor: pointer;
}

.key-container {
  padding: 5px;
  position: relative;
  font: 20px;
}

.resting {
  background-color: green;
  margin-left: 30px;
  padding: 3px 10px;
  color: #fff;
  border-radius: 5px;
}

.current {
  background-color: #83B8F4;
  margin-left: 20px;
  padding: 3px 10px;
  border-radius: 5px;
}

.steps {
  background-color: #ffcf32;
  margin-left: 20px;
  padding: 3px 10px;
  border-radius: 5px;
}

.chart-wrap {
  display: block;
  position: relative;
  margin: 30px;
  height: 400px;
  overflow-x: scroll;
}



.chart-key {
  position: absolute;
  top: 20px;
  left: 30px;
}

.chart-wrap::-webkit-scrollbar {
  -webkit-appearance: none;
  width: 7px;
}
.chart-wrap::-webkit-scrollbar-thumb {
  border-radius: 4px;
  background-color: rgba(0,0,0,.5);
  box-shadow: 0 0 1px rgba(255,255,255,.5);
}

.chart-wrap section{
  background-color: #fff;
  width: 100%;
  height: 300px;
  margin: 10px 0 0 0;
  border-radius: 10px;
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
}

.VictoryContainer {
  background-color: #fff;
  height: 90% !important; /* yes... I'm aware. This stays until I figure out Victory.js styles */
  width: 2000px !important;
  border-radius: 10px;
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  margin-top: 15px;
}

.therapies-wrap {
  position: relative;
  display: block;
  position: relative;
  margin: 40px 30px 30px;
  background-color: #fff;
  border-radius: 10px;
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
}

.therapies-wrap h2 {
  position: absolute;
  top: -27px;
  left: 0;
}

.therapies-wrap a {
  width: 50%;
  box-sizing: border-box;
  display: inline-block;
  padding: 15px;
  height: 170px;
  transition: all .5s;
  vertical-align: top;
}

.therapies-wrap a:hover {
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  cursor: pointer;
}

.therapies-wrap a img {
  display: block;
  margin: auto;
  transition: all .5s;
}

.excercise-nav {
  position: relative;
  padding: 40px;
  border-bottom: 1px solid #E4E4E4;
  text-align: center;
}

.prev,
.next {
  border: 3px solid #8f8f8f;
  font-size: 20px;
  background-color: transparent;
  margin: 0 20px 0 0;
  border-radius: 3px;
  padding: 0px 10px 2px 10px;
  text-decoration: none;
  color: #8f8f8f;
}

.breath {
  border-right: 1px solid #E4E4E4;
  border-bottom: 1px solid #E4E4E4;
  border-top-left-radius: 10px;
  text-decoration: none; 
}

.breath:hover {
  border-top-left-radius: 10px;
}

.breath:hover .lungs-icon {
  width: 130px; 
}

.lungs-icon {
  width: 120px; 
}

.breathe-wrap{
  padding: 0 0 40px 0;
  background-color: #fff;
  position: relative;
  margin: 1px 0 0 0;
  height: 600px;
  padding-bottom: 200px;
}

.mind {
  border-bottom: 1px solid #E4E4E4;
  text-decoration: none; 
}

.mind:hover {
  border-top-right-radius: 10px;
}

.therapies-wrap a .mindfulness-icon {
  width: 100px; 
  margin-top: 10px;
}

.mind:hover .mindfulness-icon {
  width: 110px; 
}

.sound {
  border-right: 1px solid #E4E4E4;
  border-bottom: 1px solid #E4E4E4;
  text-decoration: none; 
}

.sound-icon {
  width: 100px; 
}

.sound:hover .sound-icon {
  width: 108px;
}

.notes {
  border-bottom: 1px solid #E4E4E4;
  text-decoration: none; 
}

.notes-icon {
  width: 80px; 
}

.notes:hover .notes-icon {
  width: 87px;
}

.sight {
  border-right: 1px solid #E4E4E4;
  text-decoration: none; 
}

.sight:hover {
  border-bottom-left-radius: 10px;
}

.therapies-wrap a .sight-icon {
  width: 120px;
  margin-top: 30px;
}

.sight:hover .sight-icon {
  width: 135px;
}

.lifeline {
  text-decoration: none; 
  vertical-align: top;
  color: #000;
}

.lifeline:hover {
  border-bottom-right-radius: 10px;
}

.lifeline .phone-icon {
  width: 80px;
  margin-top: 10px;
}

.lifeline:hover .phone-icon {
  width: 90px;
}

.video-frame {
	position: relative;
	padding-bottom: 56.25%; /* 16:9 */
	padding-top: 25px;
	height: 0;
}
.video-frame iframe {
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
}

.coming-soon, 
.loading {
  font-size: 25px;
  text-align: center;
  width: 150px; 
  margin: 60px auto;
}

@media screen and (max-width: 500px) {
  
  html{
    height: 100%;
  }

  body {
    background-color: #fff;
    background: -moz-linear-gradient(45deg, rgba(68,161,167,1) 0%, rgb(114, 255, 203) 100%);
  background: -webkit-linear-gradient(45deg, rgba(68,161,167,1) 0%,rgba(114,255,203,1) 100%);
  background: linear-gradient(45deg, rgba(68,161,167,1) 0%,rgba(114,255,203,1) 100%);
  }

  .App-header{
    background: none;
    background-color: #E5F6FF;
    box-shadow: none;
  }

  .main-wrap{
    margin-top: 1px;
  }

  .therapies-wrap {
    margin: 0;
    border-radius: 0;
  }

  .therapies-wrap h2 {
    display: none;
  }

  .therapies-wrap a .mindfulness-icon {
    width: 80px;
    margin-top: 10px;
  }

  .chart-wrap {
    margin: 0;
    height: auto;
  }

  .chart-wrap h2 {
    display: none;
  }

  .VictoryContainer {
    border-radius: 0 !important;
    margin: 0;
  }


}