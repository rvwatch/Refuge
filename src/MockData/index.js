export const mockHeartRate = [
  { time: '00:00:00', value: 57 },
  { time: '00:01:00', value: 59 },
  { time: '00:02:00', value: 58 },
  { time: '00:03:00', value: 59 },
  { time: '00:04:00', value: 59 },
  { time: '00:05:00', value: 59 }
];

export const mockChartData = [
  { x: '12:00:00 A.M.', y: 57 },
  { x: '12:01:00 A.M.', y: 59 },
  { x: '12:02:00 A.M.', y: 58 },
  { x: '12:03:00 A.M.', y: 59 },
  { x: '12:04:00 A.M.', y: 59 },
  { x: '12:05:00 A.M.', y: 59 }
];

export const mockStepsTaken = [
  { time: '00:00:00', value: 0 },
  { time: '00:01:00', value: 0 },
  { time: '00:02:00', value: 0 },
  { time: '00:03:00', value: 0 },
  { time: '00:04:00', value: 0 },
  { time: '00:05:00', value: 0 },
  { time: '00:06:00', value: 0 },
  { time: '00:07:00', value: 0 }
];

export const mockFitbitProfile = {
  user: 'Ricardo V.',
  avgSteps: 8150
};

export const restingHeartRate = '72.34';

export const user = { username: 'ricardo' };
