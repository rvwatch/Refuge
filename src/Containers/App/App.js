import React, { Component } from 'react';
import './App.css';
import { connect } from 'react-redux';
import { object, func, bool } from 'prop-types';
import { postLogout } from '../../ApiCalls/postLogout';
import { Route, NavLink, Switch, withRouter, Link } from 'react-router-dom';
import { Breath } from '../../Components/Breath/Breath';
import { Mindfulness } from '../../Components/Mindfulness/Mindfulness';
import { Sound } from '../../Components/Sound/Sound';
import { Journal } from '../../Components/Journal/Journal';
import { Videos } from '../../Components/Videos/Videos';
import { Lifeline } from '../../Components/Lifeline/Lifeline';
import { getFitbitProfile } from '../../ApiCalls/getFitbitProfile';
import { getHeartRate } from '../../ApiCalls/getHeartRate';
import { getSteps } from '../../ApiCalls/getSteps';
import { LoginContainer } from '../LoginContainer/LoginContainer';
import { RegisterContainer } from '../RegisterContainer/RegisterContainer';
import { Settings } from '../Settings/Settings';
import Main from '../Main/Main';
import * as Actions from '../../Actions/index';


export class App extends Component {
  constructor() {
    super();
    this.state = {
      loggedIn: false
    };
  }

  async componentWillReceiveProps(nextProps) {
    const { loggedIn, heartRate } = this.props;
    if (heartRate.length) {
      return;
    }
    if (loggedIn !== nextProps.loggedIn) {
      const fitbitData = await getFitbitProfile();
      const userData = {
        user: fitbitData.user.displayName,
        avgSteps: fitbitData.user.averageDailySteps
      };
      this.props.addFitBitData(userData);
      const rawHeartRate = await getHeartRate();
      const rawStepData = await getSteps();
      const stepsTaken = rawStepData['activities-steps-intraday'].dataset;
      const heartRate = rawHeartRate['activities-heart-intraday'].dataset;
      const restingHeart = rawHeartRate['activities-heart'][0].value;
      this.props.addHeartRate(heartRate);
      this.props.addRestingHeart(restingHeart);
      this.props.addStepsTaken(stepsTaken);
    }
  }

  handleLogout = () => {
    postLogout();
    this.props.logoutUser();
    this.props.history.push('/');
  };

  render() {

    const { name } = this.props.user;
    

    const loginPath = this.props.loggedIn ? (
      <Route
        exact
        path="/"
        render={() => <Main setIntervalFun={this.setIntervalFun} />}
      />
    ) : (
      <Route exact path="/" render={() => <LoginContainer />} />
    );

    const welcomeLink = this.props.loggedIn ? <Link to="/settings">{name}</Link> :  
    <div><Link to='/login'>Sign-In </Link> <Link to='/register'> New User</Link></div>;


    return (
      <section className="App">
        <header className="App-header">
          <NavLink className="logo-wrap" to="/">
            <img
              alt="Icon"
              className="logo"
              src={require('../../Assets/images/MainLogo.svg')}
            />
          </NavLink>
          <div className="account">
            {welcomeLink}<br />
            
          </div>
        </header>
        {loginPath}
        <Switch>
        <Route exact path="/register" component={RegisterContainer} />
          <Route path ='/settings' component={Settings} />
          <Route exact path="/login" component={LoginContainer} />
          <Route exact path="/breathing" component={Breath} />
          <Route exact path="/mindfulness" component={Mindfulness} />
          <Route exact path="/sound" component={Sound} />
          <Route exact path="/journal" component={Journal} />
          <Route exact path="/videos" component={Videos} />
          <Route exact path="/support" component={Lifeline} />
        </Switch>
      </section>
    );
  }
}

export const mapStateToProps = state => ({
  fitbitData: state.fitbitData,
  heartRate: state.heartRate,
  stepsTaken: state.stepsTaken,
  user: state.user,
  loggedIn: state.loggedIn
});

export const mapDispatchToProps = dispatch => ({
  addFitBitData: data => dispatch(Actions.addFitBitData(data)),
  addHeartRate: heartRate => dispatch(Actions.addHeartRate(heartRate)),
  addStepsTaken: stepsTaken => dispatch(Actions.addStepsTaken(stepsTaken)),
  addRestingHeart: restingHeart =>
    dispatch(Actions.addRestingHeart(restingHeart)),
  logoutUser: () => dispatch(Actions.logoutUser())
});

App.propTypes = {
  logoutUser: func,
  history: object,
  loggedIn: bool,
  addFitBitData: func,
  addHeartRate: func,
  addRestingHeart: func,
  addStepsTaken: func
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
