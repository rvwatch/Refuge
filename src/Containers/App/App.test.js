import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import * as Actions from '../../Actions/';
import * as mock from '../../MockData/';

describe('App', () => {
  let wrapper;
  let addFitBitData;
  let addHeartRate;
  let addStepsTaken;
  beforeEach(() => {
    addFitBitData = jest.fn();
    addHeartRate = jest.fn();
    addStepsTaken = jest.fn();
    wrapper = shallow(
      <App
        addFitBitData={addFitBitData}
        addHeartRate={addHeartRate}
        addStepsTaken={addStepsTaken}
      />,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should add fitbit data to props', () => {
    const mockfitbitData = mock.mockFitbitProfile;
    const expected = mockfitbitData;
    const mockState = { fitbitData: mockfitbitData };
    const mapped = mapStateToProps(mockState);
    expect(mapped.fitbitData).toEqual(expected);
  });

  it('should add heartRate data to props', () => {
    const mockHeartRate = mock.mockHeartRate;
    const expected = mockHeartRate;
    const mockState = { heartRate: mockHeartRate };
    const mapped = mapStateToProps(mockState);
    expect(mapped.heartRate).toEqual(expected);
  });

  it('should add stepsTaken data to props', () => {
    const mockStepsTaken = mock.mockStepsTaken;
    const expected = mockStepsTaken;
    const mockState = { stepsTaken: mockStepsTaken };
    const mapped = mapStateToProps(mockState);
    expect(mapped.stepsTaken).toEqual(expected);
  });

  it('should add user to props', () => {
    const mockUser = mock.user;
    const expected = mockUser;
    const mockState = { user: mockUser };
    const mapped = mapStateToProps(mockState);
    expect(mapped.user).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with correct params in addFitBitData', () => {
    const mockDispatch = jest.fn();
    const mockfitbitData = mock.mockFitbitProfile;
    const expected = Actions.addFitBitData(mockfitbitData);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addFitBitData(mockfitbitData);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with correct params in addHeartRate', () => {
    const mockDispatch = jest.fn();
    const mockHeartRate = mock.mockHeartRate;
    const expected = Actions.addHeartRate(mockHeartRate);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addHeartRate(mockHeartRate);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with correct params in addStepsTaken', () => {
    const mockDispatch = jest.fn();
    const mockStepsTaken = mock.mockStepsTaken;
    const expected = Actions.addStepsTaken(mockStepsTaken);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addStepsTaken(mockStepsTaken);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with correct params in addRestingHeart', () => {
    const mockDispatch = jest.fn();
    const restingHeartRate = mock.restingHeartRate;
    const expected = Actions.addRestingHeart(restingHeartRate);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addRestingHeart(restingHeartRate);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch for logoutUser action', () => {
    const mockDispatch = jest.fn();
    const expected = Actions.logoutUser();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.logoutUser();
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
