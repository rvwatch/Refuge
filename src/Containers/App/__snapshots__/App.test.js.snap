// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addFitBitData={[Function]}
    addHeartRate={[Function]}
    addStepsTaken={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="logo-wrap"
                    to="/"
          >
                    <img
                              alt="Icon"
                              className="logo"
                              src="MainLogo.svg"
                    />
          </NavLink>
          <div
                    className="account"
          >
                    <a
                              href="#"
                              onClick={[Function]}
                    >
                              Logout
                    </a>
                    <Link
                              replace={false}
                              to="/login"
                    >
                               Login 
                    </Link>
          </div>
</header>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Switch>
          <Route
                    exact={true}
                    path="/login"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/breathing"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/mindfulness"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/sound"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/journal"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/videos"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/support"
                    render={[Function]}
          />
</Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo-wrap"
              to="/"
>
              <img
                            alt="Icon"
                            className="logo"
                            src="MainLogo.svg"
              />
</NavLink>,
            <div
              className="account"
>
              <a
                            href="#"
                            onClick={[Function]}
              >
                            Logout
              </a>
              <Link
                            replace={false}
                            to="/login"
              >
                             Login 
              </Link>
</div>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <img
                alt="Icon"
                className="logo"
                src="MainLogo.svg"
/>,
              "className": "logo-wrap",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Icon",
                "className": "logo",
                "src": "MainLogo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="#"
                  onClick={[Function]}
>
                  Logout
</a>,
                <Link
                  replace={false}
                  to="/login"
>
                   Login 
</Link>,
              ],
              "className": "account",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Logout",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Logout",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": " Login ",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": " Login ",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/login"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/breathing"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/mindfulness"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/sound"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/journal"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/videos"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/support"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/breathing",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/mindfulness",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/sound",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/journal",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/videos",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/support",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="logo-wrap"
                        to="/"
            >
                        <img
                                    alt="Icon"
                                    className="logo"
                                    src="MainLogo.svg"
                        />
            </NavLink>
            <div
                        className="account"
            >
                        <a
                                    href="#"
                                    onClick={[Function]}
                        >
                                    Logout
                        </a>
                        <Link
                                    replace={false}
                                    to="/login"
                        >
                                     Login 
                        </Link>
            </div>
</header>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Switch>
            <Route
                        exact={true}
                        path="/login"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/breathing"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/mindfulness"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/sound"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/journal"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/videos"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/support"
                        render={[Function]}
            />
</Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="logo-wrap"
                to="/"
>
                <img
                                alt="Icon"
                                className="logo"
                                src="MainLogo.svg"
                />
</NavLink>,
              <div
                className="account"
>
                <a
                                href="#"
                                onClick={[Function]}
                >
                                Logout
                </a>
                <Link
                                replace={false}
                                to="/login"
                >
                                 Login 
                </Link>
</div>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": <img
                  alt="Icon"
                  className="logo"
                  src="MainLogo.svg"
/>,
                "className": "logo-wrap",
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Icon",
                  "className": "logo",
                  "src": "MainLogo.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="#"
                    onClick={[Function]}
>
                    Logout
</a>,
                  <Link
                    replace={false}
                    to="/login"
>
                     Login 
</Link>,
                ],
                "className": "account",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Logout",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " Login ",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": " Login ",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/login"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/breathing"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/mindfulness"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/sound"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/journal"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/videos"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/support"
                render={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/breathing",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/mindfulness",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/sound",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/journal",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/videos",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/support",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
